src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs

```diff
@@ -108,6 +109,19 @@ public static class AspireAzureCosmosDBExtensions
         if (serviceKey is null)
         {
             builder.Services.AddSingleton(_ => ConfigureDb());
+
+            var csBuilder = new DbConnectionStringBuilder();
+            csBuilder.ConnectionString = settings.ConnectionString;
+
+            if (csBuilder.TryGetValue("Database", out var databaseName))
+            {
+                builder.Services.AddSingleton<Database>(sp =>
```

Comment:
What if instead is a keyedSingleton that takes the DB Name? That way the caller can obtain the Database instance for the name they need?

Reply:
Or a KeyedSingleton that takes dbName + containerName and return a Container from `client.GetContainer()`.